File: apzkr-pzpi-21-3-topchii-daria/Task1-Server/Core/Models/UserModels/UserModel.cs
using System.ComponentModel.DataAnnotations;
using Core.Models.RoomModels;

namespace Core.Models.UserModels
{
    public class UserModel : BaseModel
    {
        [Required]
        [MaxLength(50)]
        public string UserName { get; set; }

        [Required]
        [MaxLength(255)]
        public string Email { get; set; }

        [Required]
        public string HashedPassword { get; set; }

        public bool IsActive { get; set; }

        public float Height { get; set; }

        public bool Sex { get; set; }

        public DateTime BirthDate { get; set; }

        public string Information { get; set; }

        public byte[] ProfilePicture { get; set; }
    }
}

File: apzkr-pzpi-21-3-topchii-daria/Task3-WebClient/src/app/models/user.model.ts
// user.model.ts
export interface UserModel {
    id: string;
    userName: string;
    email: string;
    hashedPassword: string;
    isActive: boolean;
    height: number;
    sex: boolean;
    birthDate: Date;
    information: string;
    profilePicture: string; // Assume base64 string for simplicity
  }
  